src/queue/sq.py

```diff
@@ -0,0 +1,157 @@
+from song import Song
+
+
+class SongQueue:
+    def __init__(self):
+        self.queue = []
+
+    def __repr__(self):
+        return str(self.queue)
+
+    def peek(self):
+        """
+        Finds the queue element with the highest priority
+        :return: Song with the highest priority
+        """
+        return self.queue[0]
+
+    def pop(self):
+        """
+        Removes the song with the highest priority
+        :return: the spotify song code of the removed song
+        """
+        tmp = self.peek()
+        self.queue.pop(0)
+        return tmp
+
+    def add(self, code, ip):
+        """
+        Adds a new song to the queue
+        :param code: spotify code of song to add
+        :param ip: ip of user who added this song
+        :return:
+        """
+        new_song = Song(code, 1, ip)
+        assert (new_song not in self.queue), "No duplicates allowed in queue"
```

Comment:
started thinking about assert statements and realized it's better if we just throw an error. The handler classes has built in error catching so we can freely throw exceptions wherever the fuck we want. Look at the HandlerException class for an example exception

Reply:
Sure, I can convert to errors.  Any format for how you want it?